### DISCUSSION TITLE
How do I evaluate JS and return value with Elisp?

### DISCUSSION THREAD

**Discussion by c1-g** [emacs-eaf/emacs-application-framework#836] [Q&A] [answered]
The wiki recommend this snippet to evaluate JS in Python,
```Python
text = self.buffer_widget.execute_js("get_selection();")
```
So to replicate this in elisp, I opened an `eaf-browser` buffer on a ramdom website,
select some text, and evaluate this,
``` emacs-lisp
(eaf-call-sync "call_function_with_args" eaf--buffer-id "execute_js" "get_selection();")
```
And it gave me this message,
`[EAF/browser] Cannot call function: execute_js`

So this time, I use `eval_js` instead,
``` emacs-lisp
(eaf-call-sync "call_function_with_args" eaf--buffer-id "eval_js" "get_selection();")
```

And it gave me this,
`apply: Wrong type argument: stringp, (error "\"TypeError('eval_js() takes 1 positional argument but 2 were given')\"")`

Did I do something wrong? Can you give me some guide? That would be really helpful, thanks.

**Comment by manateelazycat** [comment, marked as answer]
Upgrade EAF and make sure JavaScript function's name is attribue of window object.

And call JS function like below:
```
(eaf-call-sync "execute_js_function" eaf--buffer-id "function_name" "function_args")
```
